packages/block-library/src/image/edit.js

```diff
@@ -273,21 +273,24 @@ export function ImageEdit( {
 			return;
 		}
 
-		const file = getBlobByURL( url );
-
-		if ( file ) {
-			mediaUpload( {
-				filesList: [ file ],
-				onFileChange: ( [ img ] ) => {
-					onSelectImage( img );
-				},
-				allowedTypes: ALLOWED_MEDIA_TYPES,
-				onError: ( message ) => {
-					isTemp = false;
-					onUploadError( message );
-				},
+		// Do not use getBlobByURL as it will only work for blobs created by us.
+		// Pasting might give us a blob URL that is not in the cache.
+		window
+			.fetch( url )
+			.then( ( r ) => r.blob() )
+			.then( ( file ) => {
+				mediaUpload( {
+					filesList: [ file ],
+					onFileChange: ( [ img ] ) => {
+						onSelectImage( img );
+					},
+					allowedTypes: ALLOWED_MEDIA_TYPES,
+					onError: ( message ) => {
+						isTemp = false;
+						onUploadError( message );
+					},
+				} );
```

Comment:
Should we also `.catch` potential errors?

Reply:
What should we do on error? Revoke the blob URL and remove it from the block (which will  change it to a placeholder state)?